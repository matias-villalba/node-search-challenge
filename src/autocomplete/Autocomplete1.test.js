const Autocomplete1 = require('./Autocomplete1');
const Suggestions = require('./Suggestions');

describe('Autocomplete1', () => {
  it('should perform a case sensitive contains search', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete1(data);

    const results = autocomplete.performSearch('biology');

    expect(results.length).toEqual(15);
    expect(results[0]).toMatchSnapshot();
  });

  it('should return results that match with words that contain upper and lower case', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete1(data);

    const results = autocomplete.performSearch('Microbiology');

    expect(results.length).toEqual(13);
    expect(results[0]).toMatchSnapshot();
  });

  it('should return no results that match with the search term', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete1(data);

    const results = autocomplete.performSearch('microbiology');

    expect(results.length).toEqual(0);
  });

  it('should return no results when search term is empty', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete1(data);

    const results = autocomplete.performSearch('');

    expect(results.length).toEqual(0);
  });

  it('should return no results when search term is a string with only spaces', () => {
    const data = Suggestions.load();
    const autocomplete = new Autocomplete1(data);

    const results = autocomplete.performSearch('  ');

    expect(results.length).toEqual(0);
  });

});
