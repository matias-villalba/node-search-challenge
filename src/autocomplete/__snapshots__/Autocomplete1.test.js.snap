// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete1 should perform a case sensitive contains search 1`] = `
Object {
  "attribution": "Marjorie Kelly Cowan",
  "briefDescription": "",
  "chapterPrice": 7.99,
  "copyrightYear": 2011,
  "description": "Get to know the smallest of organisms up close and personal with Cowan’s Microbiology on Inkling. Spin dozens of 3D molecules on their axes, watch more than 80 animations of various processes and explore more than 100 enhanced figures, all exclusive to Inkling.",
  "edition": "3rd edition",
  "price": 129.99,
  "primaryCategory": "Science",
  "publisher": "McGraw-Hill",
  "subCategories": Array [],
  "subtitle": "A Systems Approach",
  "thumbnail": "data/thumbnails/book-10.jpeg",
  "title": "Microbiology",
  "url": "https://www.inkling.com/store/book/microbiology-marjorie-kelly-cowan-3rd/",
}
`;

exports[`Autocomplete1 should return results that match with words that contain upper and lower case 1`] = `
Object {
  "attribution": "Marjorie Kelly Cowan",
  "briefDescription": "",
  "chapterPrice": 7.99,
  "copyrightYear": 2011,
  "description": "Get to know the smallest of organisms up close and personal with Cowan’s Microbiology on Inkling. Spin dozens of 3D molecules on their axes, watch more than 80 animations of various processes and explore more than 100 enhanced figures, all exclusive to Inkling.",
  "edition": "3rd edition",
  "price": 129.99,
  "primaryCategory": "Science",
  "publisher": "McGraw-Hill",
  "subCategories": Array [],
  "subtitle": "A Systems Approach",
  "thumbnail": "data/thumbnails/book-10.jpeg",
  "title": "Microbiology",
  "url": "https://www.inkling.com/store/book/microbiology-marjorie-kelly-cowan-3rd/",
}
`;
